src/app/(pages)/mobility.tsx

```diff
@@ -61,60 +117,123 @@ export default function Mobility(): JSX.Element {
                 setLoadingState(LoadingState.ERROR)
             })
     }
-
     return (
         <ScrollView
             refreshControl={
-                loadingState !== LoadingState.LOADING ? (
+                loadingState !== LoadingState.LOADING &&
+                loadingState !== LoadingState.LOADED ? (
                     <RefreshControl
                         refreshing={loadingState === LoadingState.REFRESHING}
-                        onRefresh={onRefresh}
+                        onRefresh={() => {
+                            onRefresh(mobilityKind)
+                        }}
                     />
                 ) : undefined
             }
         >
-            {loadingState === LoadingState.LOADING && (
-                <View style={styles.loadingContainer}>
-                    <ActivityIndicator size="small" color={colors.primary} />
-                </View>
-            )}
-            {loadingState === LoadingState.ERROR && (
-                <View>
-                    <Text style={[styles.errorMessage, { color: colors.text }]}>
-                        {error?.message}
-                    </Text>
-                    <Text style={[styles.errorInfo, { color: colors.text }]}>
-                        An error occurred while loading the data.{'\n'}Pull down
-                        to refresh.
-                    </Text>
+            <ToggleRow
+                items={displayTypes}
+                selectedElement={mobilityKind}
+                setSelectedElement={(x) => {
+                    if (x !== mobilityKind) {
+                        onRefresh(x)
+                        toggleMobilityKind(x)
+                    }
+                }}
+            />
+            {(mobilityKind === 'Bus' || mobilityKind === 'Train') && (
+                <View style={{ height: 'auto' }}>
+                    <SelectList
```

Comment:
The text color is always black, which is why you can't read anything in dark mode.